### DISCUSSION TITLE
OrderedConsumer Usage

### DISCUSSION THREAD

**Discussion by arhoads** [nats-io/nats.go#1736] [Q&A] [answered]
I'm new to NATS and I want to create a consumer where messages are always received in order. `OrderedConsumer` seems to be what I want, but it is an ephemeral consumer and I need it to be durable. Are all the `OrderedConsumer` settings required for in-order consumption and if not what are the required settings?

From what I can tell `MaxAckPending = 1` **is** a requirement.

**Comment by Jarema** [comment, marked as answer]
Ordered Consumer is a client side construct that creates an ephemeral consumer with no-ack and, on client side, tracks the sequences. If it detects the gap, it will recreate the consumer.
It's meant to be used if you want to quickly whip through the stream (as its really fast without acks, replication and rtt-dependand max-ack-pending) and stop.

If you want long-lived processing, durable consumer with `MaxAckPending =  1` is all you need to ensure order.

**Reply by arhoads** [reply]
Thank you! The `OrderConsumer` name made me second guess what configuration is required for ordered processing.
